geomstats/geometry/general_linear.py

```diff
@@ -1,27 +1,34 @@
 """Module exposing the GeneralLinear group class."""
 
-from itertools import product
-
+import geomstats.algebra_utils as utils
 import geomstats.backend as gs
 from geomstats.geometry.base import OpenSet
 from geomstats.geometry.lie_group import MatrixLieGroup
 from geomstats.geometry.matrices import Matrices
 
 
 class GeneralLinear(MatrixLieGroup, OpenSet):
-    """Class for the general linear group GL(n).
+    """Class for the general linear group GL(n) and its identity component.
+
+    If `positive_det=True`, this is the connected component of the identity,
+    i.e. the space of matrices with positive determinant.
 
     Parameters
     ----------
     n : int
         Integer representing the shape of the matrices: n x n.
+    positive_det: bool
+        Whether to restrict to the identity connected component of the
+        general linear group, i.e. matrices with positive determinant.
+        Optional, default: False.
     """
 
-    def __init__(self, n, **kwargs):
+    def __init__(self, n, positive_det=False, **kwargs):
```

Comment:
Why not True by default?

Reply:
So that GeneralLinear is actually the whole of GeneralLinear by default